hikari/messages.py

```diff
@@ -1267,10 +1267,10 @@ async def respond(
         builtins.TypeError
             If both `attachment` and `attachments` are specified.
         """  # noqa: E501 - Line too long
-        if reply is True:
+        if reply:
             reply = self
 
-        elif reply is False:
+        elif not reply:
```

Comment:
These seperate `bool` from `undefined.UNDEFINED`, so this actually creates erroneous behaviour